src/EthereumHelpers.js

```diff
@@ -93,27 +94,32 @@ function readEventFromTransaction(
  *         object once it is received.
  */
 function getEvent(sourceContract, eventName, filter) {
-  return new Promise((resolve, reject) => {
+  return new Promise(async (resolve, reject) => {
     // As a workaround for a problem with MetaMask version 7.1.1 where subscription
     // for events doesn't work correctly we pull past events in a loop until
     // we find our event. This is a temporary solution which should be removed
     // after problem with MetaMask is solved.
     // See: https://github.com/MetaMask/metamask-extension/issues/7270
-    const handle = setInterval(
-      async function() {
+    await pWaitFor(
+      async () => {
         // Query if an event was already emitted after we start watching
-        const event = await getExistingEvent(sourceContract, eventName, filter)
+        let event
+        try {
+          event = await getExistingEvent(sourceContract, eventName, filter)
+        } catch (error) {
+          console.warn(`failed to get existing event: ${error.message}`)
+        }
 
         if (event) {
-          clearInterval(handle)
           resolve(event)
+          return true
         }
+        return false
```

Comment:
I guess if `sourceContract.once` resolves the event here, this will run until it also manages to resolve the event, and that second `resolve` will just be a noop.